### PR TITLE
Inbuilt lexer

### LINKED ISSUE #9: Create a new lexer

### PR DESCRIPTION
**Pull request by ikezedev**
We could not get logos to fit all of our needs and the regex was really getting out of hands already. Resolves #9 

- [x] whitespace and comments
- [x] keywords and operators
- [x] literals
    - [x] Int
    - [x] Hex
    - [x] Bin
    - [x] Long
    - [x] Null
    - [x] Boolean
    - [x] Char  
- [x] Identifiers
- [x] Strings
    - [x] single line
    - [x] multiline

### CODE REVIEW

File: crates/tokenizer/src/tokens_temp.rs
Comment (ikezedev) [inline crates/tokenizer/src/tokens_temp.rs:1]: Will remove this file at the end. using it only for reference now

**Review by iambenkay — DISMISSED** [review]
LGTM

**Comment by iambenkay** [comment]
Will add some tests to this PR to see if I can expose any edge cases

**Review by iambenkay — APPROVED** [review]
LGTM. Great work on the String parsing functionality in the lexer

**Pull request merged**